### PR TITLE
Update developer docs

### PR DESCRIPTION
**Pull request by bkjohnson**
Related to https://github.com/department-of-veterans-affairs/va.gov-team/issues/23328

This updates the developer documentation

### CODE REVIEW

File: src/_documentation/for-developers.md [hunk 1]
```diff
@@ -3,29 +3,22 @@ layout: default
...
 **If you are working in the vets-website repository**, you can skip straight to the [developer documentation](https://department-of-veterans-affairs.github.io/veteran-facing-services-tools/). Otherwise, proceed below.
 
-<div class="vads-u-background-color--gold vads-u-padding--2 vads-u-display--inline-block vads-u-width--auto">
-  <p class="vads-u-margin--0  vads-u-measure--5"><strong>We are still improving how you can install Formation into your project. In the meantime, these instructions should help get you to get started with what is currently available.</strong></p>
-</div>
-
-
-
-How you implement Formation into your project depends on how your project is structured and your preferences. The easiest way to get started is by using `npm`. We will have a direct download available in the future.
+How you implement VADS styles into your project depends on how your project is structured and your preferences. The easiest way to get started is by using `npm`. We will have a direct download available in the future.
```
Comment (cvalarida) [inline src/_documentation/for-developers.md:21]: > We will have a direct download available in the future.

Is this true? I think we can remove this until we decide it's a thing we actually want to do.

File: src/_documentation/for-developers.md [hunk 2]
```diff
@@ -65,15 +43,40 @@ $formation-font-path: "#{$formation-asset-path}/fonts";
...
+
+### Load the Web Component library
 
-If you would like to use the Sass functions, such as for [spacing](../design/spacing-units.html#using-the-spacing-tokens), and variables in your project, you can either move the necessary files with gulp, or import the files from your project scss. This depends greatly on your project structure. This documentation site imports Formation’s Sass files in its [application.scss](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/blob/master/src/assets/stylesheets/application.scss#L5).
+The Design System team is working on developing a library of reusable Web Components that can be used on any HTML page. Once loaded on the page, these can be used like normal HTML tags in JSX without the need to import any additional code.
```
Comment (cvalarida) [inline src/_documentation/for-developers.md:52]:
```suggestion
The Design System team is working on developing a library of reusable Web Components that can be used on any HTML page or React application.
```

File: src/_documentation/for-developers.md [hunk 3]
```diff
@@ -65,15 +43,40 @@ $formation-font-path: "#{$formation-asset-path}/fonts";
...
 
-## CSS
+This is already handled for the `vets-website` repository. To get our Web Component library set up in a new project, here is what we recommend:
```
Comment (cvalarida) [inline src/_documentation/for-developers.md:54]: This section is really clear. Nicely done! :raised_hands:

File: src/_documentation/for-developers.md [hunk 4]
```diff
@@ -112,7 +115,7 @@ Don’t use Sass shorthand features, such as nesting with ampersands often used
 </div>
 </div>
 
-### Modifying components
+#### Modifying components
```
Comment (cvalarida) [inline src/_documentation/for-developers.md:118]: I don't think this is a sub-section under `CSS`, is it?

File: src/_documentation/for-developers.md [hunk 5]
```diff
@@ -162,4 +165,11 @@ In general, some rules for implementing design work include:
 - Use the Formation [naming convention](naming)
 - Do not use ID selectors
 
+## Contributing to the Design System
+
+The Design System currently has an ["Experimental" Design System](../experimental-design) with less oversight which VFS application teams can contribute to. In order for a component to be included in the _official_ Design System, here is what we expect to see:
```
Comment (cvalarida) [inline src/_documentation/for-developers.md:170]: This should be covered somewhere else. That said, I want to make sure they include a link to where the code can be found when they _do_ add the documentation in `Experimental Design`.

File: src/_documentation/for-developers.md [hunk 6]
```diff
@@ -162,4 +165,11 @@ In general, some rules for implementing design work include:
...
 
+1. Research on why this new addition is useful/necessary.
+1. Guidance on how to use it.
```
Comment (cvalarida) [inline src/_documentation/for-developers.md:173]: So....yes, but no. This is true, but doesn't belong in the developer section of this documentation.

File: src/_documentation/for-developers.md [hunk 7]
```diff
@@ -162,4 +165,11 @@ In general, some rules for implementing design work include:
...
+1. 90%+ test coverage
+1. Any PR should be manageable in size (less than ~500 lines of code)
```
Comment (cvalarida) [inline src/_documentation/for-developers.md:175]: This is the part I'd like to keep. The audience for this document is developers, and this is the developer-facing part. :+1:

**Review by cvalarida — COMMENTED** [review]
I think there's still work to be done in the `Contributing to the Design System` section, but the rest looks good. We can talk about what we should put in that section either here or in Slack, though.

File: src/_documentation/for-developers.md (hunk 6 above)
Comment (bkjohnson) [reply src/_documentation/for-developers.md:173]: I was thinking the same thing and a bit hesitant about including it. I'll :axe: it.

File: src/_documentation/for-developers.md [hunk 8]
```diff
@@ -162,4 +165,11 @@ In general, some rules for implementing design work include:
 - Use the Formation [naming convention](naming)
 - Do not use ID selectors
 
+## Contributing to the Design System
+
+If you contribute something to the Experimental Design System, we would like for a link to that code (in `vets-website`) to be provided in the documentation.
```
Comment (cvalarida) [inline src/_documentation/for-developers.md:170]:
```suggestion
If you contribute something to the Experimental Design System, include a link to that code (in `vets-website`) in the documentation.
```

File: src/_documentation/for-developers.md [hunk 9]
```diff
@@ -162,4 +165,28 @@ In general, some rules for implementing design work include:
...
 - Do not use ID selectors
 
+## Contributing to the Design System
+
+The two main ways for developers to contribute to the Design System are by writing new components or by modifying existing components. Regardless of which type of contribution you are making, we expect each PR to:
+
+- have at least 90% test coverage
+- have appropriate comments/documentation for functions, classes, etc.
+- have Storybook stories for new features
+- minimize complexity - don't build for features which aren't in the system yet
```
Comment (cvalarida) [inline src/_documentation/for-developers.md:175]:
```suggestion
- minimize complexity
- include only the smallest changeset required for the feature or fix
```

File: src/_documentation/for-developers.md [hunk 10]
```diff
@@ -162,4 +165,28 @@ In general, some rules for implementing design work include:
...
+In order for a component to be included in the _official_ Design System, we expect [these criteria to be met]().
+-->
+
+
+### Modifying existing code
+
+PRs which make a change to the Design System should be manageable in size (less than ~500 lines of code). This is to meant to:
 
+- Save your time as the developer
+- Help prevent unnecessary things from making their way into the Design System
```
Comment (cvalarida) [inline src/_documentation/for-developers.md:191]:
```suggestion
- Keep PRs tightly focused
```

File: src/_documentation/for-developers.md [hunk 11]
```diff
@@ -3,29 +3,22 @@ layout: default
...
 **If you are working in the vets-website repository**, you can skip straight to the [developer documentation](https://department-of-veterans-affairs.github.io/veteran-facing-services-tools/). Otherwise, proceed below.
 
-<div class="vads-u-background-color--gold vads-u-padding--2 vads-u-display--inline-block vads-u-width--auto">
-  <p class="vads-u-margin--0  vads-u-measure--5"><strong>We are still improving how you can install Formation into your project. In the meantime, these instructions should help get you to get started with what is currently available.</strong></p>
-</div>
-
-
-
-How you implement Formation into your project depends on how your project is structured and your preferences. The easiest way to get started is by using `npm`. We will have a direct download available in the future.
+How you implement VADS styles into your project depends on how your project is structured and your preferences. The easiest way to get started is by using `npm`.
```
[the remark this answers is not in the document]
Comment (bkjohnson) [reply src/_documentation/for-developers.md:21]: I don't see any real harm in it - I'll add it with the caveat that it should only be for prototyping.

**Pull request merged**